error[E0119]: conflicting implementations of trait `std::marker::Unpin` for type `Foo<_, _>`:
  --> $DIR/unpin_conflict.rs:10:1
   |
10 | #[pin_projectable] //~ ERROR E0119
   | ^^^^^^^^^^^^^^^^^^ conflicting implementation for `Foo<_, _>`
...
26 | impl<T, U> Unpin for Foo<T, U> where T: Unpin {} // Conditional Unpin impl
   | --------------------------------------------- first implementation here

error[E0119]: conflicting implementations of trait `std::marker::Unpin` for type `Bar<_, _>`:
  --> $DIR/unpin_conflict.rs:30:1
   |
30 | #[pin_projectable] //~ ERROR E0119
   | ^^^^^^^^^^^^^^^^^^ conflicting implementation for `Bar<_, _>`
...
46 | impl<T, U> Unpin for Bar<T, U> {} // Non-conditional Unpin impl
   | ------------------------------ first implementation here

error[E0119]: conflicting implementations of trait `std::marker::Unpin` for type `Baz<_, _>`:
  --> $DIR/unpin_conflict.rs:48:1
   |
48 | #[pin_projectable] //~ ERROR E0119
   | ^^^^^^^^^^^^^^^^^^ conflicting implementation for `Baz<_, _>`
...
64 | impl<T: Unpin, U: Unpin> Unpin for Baz<T, U> {} // Conditional Unpin impl
   | -------------------------------------------- first implementation here

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0119`.
